include/zephyr/lorawan/lorawan.h

```diff
@@ -371,6 +371,20 @@ int lorawan_clock_sync_get(uint32_t *gps_time);
 
 #endif /* CONFIG_LORAWAN_APP_CLOCK_SYNC */
 
+#ifdef CONFIG_LORAWAN_REMOTE_MULTICAST
+
+/**
+ * @brief Run Remote Multicast Setup service
+ *
+ * This service is responsible for multicast session key exchange and setting
+ * up a class C session. The keys are stored in the non-volatile memory.
+ *
+ * @return 0 if successful, negative errno otherwise.
+ */
+int lorawan_remote_multicast_run(void);
```

Comment:
Why can't this be run automatically in `SYS_INIT(POST_KERNEL, 1)`? From what I can tell it is only initialising memory, and the downlink list init is run in `SYS_INIT(POST_KERNEL, 0)`.

Reply:
The idea was that the LoRaWAN services should only be run after the network was joined successfully, so the setup/starting of the service must be controlled by the application. As far as I know, we don't have a LoRaWAN API to determine if the device has successfully joined a network apart from the return code of the `lorawan_join` call.

Reply:
But that function doesn't schedule any transmissions or do anything that relies upon being connected.
It registers for downlink callbacks, and initialises two `k_work` structures.

Reply:
Yeah, it would still react on incoming frames and do things like switching to class C. I thought it would be good to leave it to the application to enable the function. But it's probably fair to assume that when you compile multicast setup in you also definitely want to use it.

Updated to use `SYS_INIT` now.